apis/v1alpha1/tcproute_types.go

```diff
@@ -51,13 +51,15 @@ type TCPRoute struct {
 
 // TCPRouteRule is the configuration for a given rule.
 type TCPRouteRule struct {
-	// Match defines which connections match this rule.
+	// Matches define conditions used for matching the rule against
+	// incoming TCP connections.
+	// Each match is independent, i.e. this rule will be matched
+	// if **any** one of the matches is satisfied.
 	//
 	// +optional
-	Match *TCPRouteMatch `json:"match" protobuf:"bytes,1,opt,name=match"`
+	Matches []TCPRouteMatch `json:"matches" protobuf:"bytes,1,rep,name=matches"`
```

Comment:
Although I like pluralizing matches for HTTPRoute it's less clear what the use case is here. I believe the only kind of match we support at this level is an `extensionRef`. Since it's always easy to move from singular to plural in the future I think I'd personally rather leave this as singular until we have a use case for multiple matches here.

Reply:
The background for this change is in the following conversation:
https://github.com/kubernetes-sigs/service-apis/pull/277#discussion_r476107045

This change is more about API consistency than anything else.

Reply:
Yeh, I'd prefer to be consistent with the plurality even though the use case is multiple extension matches.

Reply:
@robscott Do you agree/disagree here?
(Pinging to move the PR forward to merge or close)

Reply:
👍 I can get behind that for the sake of consistency

Reply:
Okay, resolving this thread on this PR then.